### ISSUE TITLE
Unable to set url prefix for the application

### ISSUE THREAD

**Issue by adam-delezuch-goodc** [talentlessguy/tinyhttp#238]
[<img alt="Issuehunt badges" src="https://img.shields.io/badge/IssueHunt-%245%20Rewarded-%237E24E3.svg" />](https://issuehunt.io/r/talentlessguy/tinyhttp/issues/238)

The application model provides for several api versions for one application. Setting the url prefix is impossible because it selects the first element in the array

```javascript
import { App } from '@tinyhttp/app'

const routes = [
  '/products',
  '/search',
  '/basket'
]

app
  .use('api/v1', [...routes])
  .listen(PORT, () => {
})
```

**Versions**

* `node`: 14
* `@tinyhttp/app`: 1.1.13

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $5.00)

- [<img src='https://avatars.githubusercontent.com/u/35937217?v=4' alt='talentlessguy' width=24 height=24> talentlessguy](https://issuehunt.io/u/talentlessguy) ($5.00)

### Submitted pull Requests
- [#249 fix: Unable to set url prefix for the application](https://issuehunt.io/r/talentlessguy/tinyhttp/pull/249)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/talentlessguy/tinyhttp/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

**Comment by talentlessguy** [comment]
Could you try with

```js
app.use('/api/v1', ...routes)
```

?

**Comment by adam-delezuch-goodc** [comment]
This code is not working.

See my example:
```javascript
// upload.js
import { App } from '@tinyhttp/app'
const router = new App()
router.get('/upload', (req, res) => res.send('upload'))
export { router as upload }
```

```javascript
// product.js
import { App } from '@tinyhttp/app'
const router = new App()
router.get('/product', (req, res) => res.send('product'))
export { router as product }
```

```javascript
// routes.js
import { product } from './product.js'
import { upload } from './upload.js'
export default [ products, upload ] reverse this array
```

```javascript
// server.js
import { App } from '@tinyhttp/app'
import { default as routes } from './routes.js'
const app = new App()

app
  .use('api/v1/', ...routes)
  .listen(PORT, () => {})
```

In this case, the product will return status 200 and upload status 404.

**Comment by talentlessguy** [comment]
Looks like something serious and probably I broke it with a recent commit

will try to fix asap

**Comment by talentlessguy** [comment]
I think the problem occurs here: https://github.com/talentlessguy/tinyhttp/blob/9a5573fb7fd724d52217745c839d81bacb3f90f9/packages/app/src/app.ts#L109

And sadly I don't know how to solve the problem.... :c

If you have an idea of how to fix that I would be very awesome to share

From what I understand the problem is that the first app gets mounted and then the apps coming after don't get handled (e.g. it gets stuck only on the first app in the queue)

**Comment by talentlessguy** [comment]
It turns out to be a bug in Polka: https://github.com/lukeed/polka/issues/159

**Comment by talentlessguy** [comment]
A temporary fix would be this:

```js
app.use('/api/v1/app1', ...)
app.use('/api/v1/app2', ...)
```

**Comment by talentlessguy** [comment]
the issue got fixed (thanks to @wasd845)

today I will release a minor patch with the fix

**Comment by talentlessguy** [comment]
closed via b49b999f11b6de775dcd807f1702a461616576c5

the fix landed in tinyhttp v1.3.3! :tada: 

```sh
pnpm i @tinyhttp/app@1.3.3
```
